src/schema/v2/partner/partnerAlerts.ts

```diff
@@ -0,0 +1,101 @@
+import { pageable } from "relay-cursor-paging"
+import {
+  GraphQLString,
+  GraphQLObjectType,
+  GraphQLList,
+  GraphQLBoolean,
+} from "graphql"
+import {
+  connectionWithCursorInfo,
+  paginationResolver,
+} from "schema/v2/fields/pagination"
+import { convertConnectionArgsToGravityArgs } from "lib/helpers"
+import { CollectorProfileType } from "schema/v2/CollectorProfile/collectorProfile"
+
+const PartnerCollectorProfilesConnectionType = connectionWithCursorInfo({
+  name: "PartnerCollectorProfiles",
+  nodeType: CollectorProfileType,
+}).connectionType
+
+export const PartnerAlertType = new GraphQLObjectType({
+  name: "PartnerAlert",
```

Comment:
New concept! We have an 'alert' abstraction already -> so what is a 'partner alert'? If it's all just alerts all the way down at the end of the day - this adds a new abstraction that might not be needed (better to use 'Alert').

An alternate schema possibly might be a connection between a partner and alerts (so a connection with the node being an alert), with some of the partner-specific stuff (like 'score') on the edge.

Reply:
I agree that 'partner alert' and 'alert' share many similarities, particularly in terms of metadata like attribution class and price range. However, there are _some_ differences that I believe necessitate a distinction, such as handling the scoring on individual nodes. (This involves determining which alert is the best, whether the partner has already satisfied this alert by uploading an artwork, and soon, sharing an artwork with users subscribed to the alert. etc etc)

Your suggestion of combining the Alert type with some partner-specific fields could indeed address both aspects effectively. I'll experiment with this idea and sketch out a schema that incorporates these considerations!!

Reply:
Ah I think I understand now, I was thinking attributes on the edge would be able the "collection" of nodes but its more about the relationship between partner <> alert

Ex:
```
{
  "data": {
    "partner": {
      "id": "1",
      "name": "Partner A",
      "alerts": {
        "edges": [
          {
            "node": {
              "id": "101",
              "createdAt": "2024-07-09T12:00:00Z"
            },
            "score": 85,
            "cursor": "cursor1"
          },
          {
            "node": {
              "id": "102",
              "createdAt": "2024-07-09T13:00:00Z"
            },
            "score": 90,
            "cursor": "cursor2"
          }
        ],
        "pageInfo": {
          "endCursor": "cursor2",
          "hasNextPage": true
        }
      }
    }
  }
}
```

Reply:
Ok @mzikherman Ive noted that I owe you some bagels again in my TODOs.

I've adjusted this to return `AlertType` as the base node and the data from partner search criteria will be on the edges

```
{
  partner(id: "commerce-test-partner") {
    slug
    name
    partnerAlertsConnection(first: 5) {
      totalCount
      edges {
        partnerId
        matchedAt
        score
        node {
          internalID
          colors
          additionalGeneNames
          formattedPriceRange
        }
      }
    }
  }
}
```

Which returns lots of nice stuff:
```
{
  "data": {
    "partner": {
      "slug": "commerce-test-partner",
      "name": "Commerce Test Partner",
      "partnerAlertsConnection": {
        "totalCount": 10,
        "edges": [
          {
            "partnerId": "5f80bfefe8d808000ea212c1",
            "matchedAt": null,
            "score": null,
            "node": {
              "internalID": "f7ed40ab-93d2-4e3a-82c1-025b7feb4331",
              "colors": [],
              "additionalGeneNames": [
                "Prints"
              ],
              "formattedPriceRange": null
            }
          },
          {
            "partnerId": "5f80bfefe8d808000ea212c1",
            "matchedAt": null,
            "score": null,
            "node": {
              "internalID": "2ad7213d-bbbb-4082-8eeb-e2f14d4ea64f",
              "colors": [],
              "additionalGeneNames": [
                "Painting"
              ],
              "formattedPriceRange": null
            }
          },
          {
            "partnerId": "5f80bfefe8d808000ea212c1",
            "matchedAt": null,
            "score": null,
            "node": {
              "internalID": "5ba2a417-7d7b-4728-a653-9c6b3dc96d7e",
              "colors": [],
              "additionalGeneNames": [
                "Drawing",
                "Painting",
                "Work on Paper"
              ],
              "formattedPriceRange": "$1,000 - 5,000"
            }
          },
          
```

Let me know if that aligns with your thinking 
(I think Im going to pull out the collector profile work into a separate PR)

Reply:
Yes! I suggested it a bit hastily (potentially - didn't necessarily mean for you to _definitely_ go that route - but it seems like it was fruitful) - and more automatically.

What I mean is that - a connection with data on the edges is a nice setup for join models, and 'synthetic'-ish models of that kind. So if we have a concept of 'type A', and 'type B', if someone adds 'type AB' or 'type BA' - that's an automatic signal to me to suggest A -> B (and put stuff on the edge), or B -> A (and put stuff on the edge).

So here, we have `Alert` as a basic type, and we have `Partner` as well. On the backend, `Alert` is heavily based on `SearchCriteria`, _and_ we now have `PartnerSearchCriteria` as a thing.

Whew! All that is to say, something like `PartnerAlert` or `PartnerSearchCriteria`, _at least in schema terms_ -> to me sounded like one of those join-esque cases where edges can be a natural home and avoiding exposing (to clients), this sort-of implementation-y concept (of `PartnerAlert` or `PartnerSearchCriteria`).

However - of course there's no real right or wrong answer here - there's some science but there's an art as well - if this feels like a nice schema design lets do it!

Reply:
Also, potentially keep this in mind for exposing of collector profiles. You'd want the node to be `CollectorProfile`, but some of that 'extra' partner-specific data that the [Gravity endpoint exposes](https://github.com/artsy/gravity/blob/aec07cc1753226ba7b9f9bcb5fa3b78c1bf4a9a9/app/presenters/partner_collector_profile_presenter.rb#L13-L15) - if you plan on adding to it or showing it in the UI - could be along the edge. Same kind of setup with `resolveNode: (resp) => resp.collector_profile` and so forth

Reply:
++ yes thanks for the deeper dive, I think I agree! This feels better once in the fingers and reviewing [this article ](https://www.apollographql.com/blog/explaining-graphql-connections)you've shared in the past!

Reply:
I think I'll fast follow with `CP` stuff, want to play around a little more with it!